\chapter{Conclusion}
\label{chap:conclusion}

\section{Summary}

The overarching scientific question at the root of the work I conducted during my PhD is the following: how can evolution, an inherently random process, sometimes be reproducible?
In order to refine this far-reaching problem into a more actionable research program, I focused on the study of the epistatic interactions between different kinds of mutations, and asked the following question: can an improved understanding of the beneficial or deleterious nature of epistatic interactions help us predict which strain within a given population has the highest potential to harbor future favorable mutations and outcompete other strains, given its current genetic background?
Inspired by results from the \emph{Long-Term Evolution Experiment}, I singled out DNA supercoiling as a prime example with which to try to corroborate this suggestion (Chapter~\ref{chap:background}).
The level of DNA supercoiling is indeed at the same time intricately governed by many cellular processes -- that are subject to possible mutations -- and a fundamental actor in the regulation of gene transcription and expression.
The central role of DNA supercoiling, which bridges the physical structure of the genome with the molecular phenotype of the cell, makes mutations that influence its level prime suspects in the shaping of evolutionary trajectories by epistatic interactions.

In order to tackle this problem, I first implemented a simple model of the level of supercoiling and its effect on transcription in the \emph{Aevol} \emph{in silico} experimental evolution platform, but could not establish a measurable role of supercoiling mutations in speeding up evolution after they occur (Chapter~\ref{chap:aevol}).
As I estimated this lack of results to be due to the inner complexity of the \emph{Aevol} model and to the overly simplistic representation of supercoiling that I was able to incorporate into \emph{Aevol}, I designed and implemented a new multi-scale model, called \emph{EvoTSC}.
This new model trades off the precise genome description of \emph{Aevol} for a much more detailed model of the coupling between transcription and supercoiling.
I first validated the relevance of this new model by showing that its description of supercoiling is rich enough to allow for the evolution of differentiated expression patterns in response to environmental perturbations (Chapter~\ref{chap:alife}), even when the only mutations are genomic inversions, and observed the emergence of relaxation-activated genes in the model (Chapter~\ref{chap:ploscb}).
I then characterized the gene regulatory networks, mediated by the transcription-supercoiling coupling, that are at the root of these differentiated transcriptional responses.
Furthermore, I showed that these regulatory networks are based in the relative positions of genes on the genome, and that they require the interaction of multiple genes to function, spanning wide swathes of the genome.

In order to reinforce the degree of confidence that we can have in the conclusions hitherto drawn from the \emph{EvoTSC} model, I then ran additional sets of simulations using the model with different parameter values (Chapter~\ref{chap:param}).
I showed that these results are robust with respect to the size of the topological domains (the distance at which supercoiling propagates), to the size of the intergenic regions within the range of observed bacterial values, and to the intensity of the environmental perturbations.
Overall, I was able to make two main conclusions regarding the transcription-supercoiling coupling with the help of the \emph{EvoTSC} model.
First, I showed that this coupling provides a material basis for gene regulation in the absence of transcription factors, as well as a mechanism that could explain the activation of certain genes by DNA relaxation.
I then showed that this coupling also plays a plausible role in the shaping of the organization of bacterial genomes over evolutionary time, via the supercoiling-mediated gene regulatory networks that evolve through successive genomic rearrangements.

These results, even though they provide some insight into the evolution of bacterial genomes, do however not yet answer the original question of the epistatic interactions between supercoiling mutations and other mutations, which was left inconclusively answered in Chapter~\ref{chap:aevol}.
I therefore re-attacked this question using the \emph{EvoTSC} model, by introducing supercoiling mutations along the original genomic inversions, and studying the associated supercoiling fitness landscapes (Chapter~\ref{chap:epistasis}).
I first showed that populations in which supercoiling can mutate along genomic inversions evolve a slightly higher fitness after the same number of generations than populations in which it cannot.
Then, replicating the environmental shock at the beginning of the \emph{LTEE} into the \emph{EvoTSC} model, I showed that populations with supercoiling mutations also recover faster from an environmental shock than populations without supercoiling mutations.
Finally, I showed that on their own, supercoiling mutations are only able to provide a local exploration of the supercoiling fitness landscape that stems from the organization of genes on the genomes, but that they do so in a repeatable manner despite their inherent randomness.

\paragraph{} In this work, I used an \emph{in silico} experimental evolution approach to study the evolution of gene regulation by DNA supercoiling in bacteria.
Using both the \emph{Aevol} and \emph{EvoTSC} platforms, I showed that supercoiling mutations seem to be selected for their direct fitness benefits, rather than for the increased evolvability that they could confer on their bearer through biased epistatic interactions.
These results indicate that the repeated supercoiling mutations observed in the \emph{LTEE} could similarly have been selected as a result of their direct benefits.

The results that I obtained with the \emph{EvoTSC} model nonetheless underscore the significant role that supercoiling could play in gene regulation in bacteria.
Indeed, these results show that, far from being limited to the direct biophysical effect of DNA supercoiling on transcription, gene regulatory networks that are expressive enough to activate or inhibit subsets of genes can evolve when the only regulator of gene expression is the supercoiling generated by gene transcription.
In particular, regulation by supercoiling is sufficient to generate relaxation-activated genes, such as the ones found in \emph{E. coli}, \emph{S. pneumoniae} or \emph{D. dadantii}.
It could moreover explain the evolutionary conservation of the relative positions of groups of neighboring genes, or syntenies, that has been for example observed between \emph{E. coli} and \emph{S. enterica}, as these groups of genes could coordinate their expression levels through local changes in supercoiling.
Finally, these results reinforce the hypothesis that DNA supercoiling could play an important regulatory role in streamlined bacteria such as \emph{B. aphidicola}, whose genome is nearly devoid of traditional transcription factors.


\section{Perspectives}

The work presented in this manuscript could be furthered along at least two main directions.
The first one would be to continue the investigation of supercoiling as a representative example of the role that epistatic interactions play in the structure of fitness landscapes, in order to better understand the repeatability of the apparition and fixation of such mutations in the \emph{LTEE}.
A first step in this direction would be to pursue in more detail the analysis of evolutionary trajectories in the \emph{EvoTSC} model, by recording individual lineages during evolution and explicitly reconstructing the associated mutational histories.
This would allow the same experiment as the one presented in \emph{Aevol} (in Chapter~\ref{chap:aevol}) to be performed in a model in which supercoiling mutations have an effect on the fitness landscape -- and hence on evolutionary trajectories -- that is less independent from the rest of the genotype, and in particular from genome structure.
Studying the fixation times of mutations in the \emph{EvoTSC} model could in particular provide data that is easier to interpret than in \emph{Aevol}, as there are only two possible kinds of mutations (genomic inversions and supercoiling mutations) in \emph{EvoTSC}.
Another natural, but slightly more difficult to carry out, step in that direction would be to instead implement the more precise model of supercoiling used in \emph{EvoTSC} in \emph{Aevol}.
Replicating the results already obtained in \emph{EvoTSC} in a model in which genomes can evolve in additional ways in response to supercoiling mutations compared to \emph{EvoTSC} would therefore provide an additional degree of confidence in these results.

The common thread at the root of both these initiatives is to build models in which supercoiling mutations are sufficiently finely modeled to allow for alterations of the fitness landscape by these mutations that are substantial enough to guide evolution towards different paths conditionally to their occurrence in a lineage, and then determine whether they observably do so.
Whether in \emph{EvoTSC} or in \emph{Aevol}, it seems promising to explore this direction more quantitatively by computing the distribution of fitness effects of supercoiling mutations in either model or the fitness of double mutants (coupling a genomic inversion with a supercoiling mutation), instead of relying on the random occurrence of these mutations in evolving populations.
This more local -- but more detailed -- analysis of the fitness landscape could shed further light on the possible bias of supercoiling mutations towards beneficial epistatic interactions with other mutations.

The second main direction in which to pursue the work presented in this manuscript would be moving away from questioning the evolutionary role of supercoiling and instead towards a quantitative description of the regulatory role of the transcription-supercoiling coupling, at the mesoscale of bacterial topological domains.
As supercoiling has been hypothesized to play an important role in bacterial gene regulation~\citep{elhoudaigui2019}, it would be very interesting to obtain a model that explicitly takes it into account when predicting the expression levels of genes in a given, possibly synthetic, genetic system.
One of the main drawbacks of the model of the interaction between transcription and supercoiling that is used in \emph{EvoTSC} in this regard is its lack of an explicit time scale.
As such, \emph{EvoTSC} therefore models the expected average behavior of genes subjected to this interaction, but cannot adequately picture the dynamic and stochastic nature of gene transcription by RNA polymerases.
Yet, this stochasticity plays an important role in the accurate modeling of gene transcription dynamics, as exemplified in~\cite{sevier2021}.
Incorporating an explicit description of the movement of polymerases along DNA, of the resolution of supercoils by topoisomerases, and of the formation of supercoiling barriers by nucleoid-associated proteins into \emph{EvoTSC} would provide us with a model with which to disentangle the contribution of these processes to gene transcription and provide quantitative predictions of gene expression levels.
Another recently evidenced component of the effect of DNA supercoiling on bacterial gene transcription is the effect of promoter discriminator sequence and spacer length on transcription initiation~\citep{forquet2021,forquet2022,pineau2022a}.
Incorporating this neglected effect into the model should reinforce the quality of its prediction of gene expression levels.
Coupled with the evolutionary model of \emph{EvoTSC}, such a quantitative model would allow the study of the transcriptional response of genetic systems to perturbations caused by mutations in each of their components, be it gene order, promoter sequence, or topoisomerase activity, and hence the prevision of possible future evolutionary trajectories.
