\chapter{Conclusion}
\label{chap:conclusion}

\section{Summary}

The overarching scientific question at the root of the work I conducted during my PhD is the following: can evolution, a supposedly random process, be made predictable?
In order to refine this far-reaching endeavor into a more actionable research program, I focused on the study of the epistatic interactions between mutations, and asked the following question: can an improved understanding of the beneficial or deleterious nature of epistatic interactions help us predict which strain has the highest potential to harbor future favorable mutations and overtake competing strains, given its current genetic background?
Inspired by results from the \emph{Long-Term Evolution Experiment}, I singled out DNA supercoiling as a prime example with which to try to corroborate this suggestion (Chapter~\ref{chap:background}).
The level of DNA supercoiling is indeed at the same time intricately governed by many cellular processes, that are subject to possible mutations, and a fundamental actor in the regulation of gene transcription and expression.
The central role of DNA supercoiling, which bridges the physical structure of the genome with the molecular phenotype of the cell, makes mutations that influence its level prime suspects in the shaping of evolutionary trajectories by epistatic interactions.

In order to tackle this problem, I first implemented a simple model of the level of supercoiling and its effect on transcription in the \emph{Aevol} \emph{in silico} experimental evolution platform, but could not establish a measurable role of supercoiling mutations in speeding up evolution after their appearance (Chapter~\ref{chap:aevol}).
As I estimated this lack of results to be due to the inner complexity of the \emph{Aevol} model and to the initially extremely simple representation of supercoiling that I was able to incorporate into it, I designed and implemented a new multi-scale model, \emph{EvoTSC}, which trades off the precise genome description of \emph{Aevol} for a much more detailed model of the coupling between transcription and supercoiling.
I first validated the relevance of this new model by showing that its description of supercoiling is rich enough to allow for the evolution of differentiated expression patterns in response to environmental perturbations (Chapter~\ref{chap:alife}), even when the only mutations are genomic inversions, and observed the emergence of relaxation-activated genes in the model (Chapter~\ref{chap:ploscb}).
I then characterized the gene regulatory networks, mediated by the  transcription-supercoiling coupling, that are at the root of these differentiated transcriptional responses.
I showed that these regulatory networks are based in the relative positions of genes on the genome, and that they require the interaction of multiple genes to function, spanning wide swaths of the genome.
In order to reinforce confidences in the conclusions hitherto drawn from the model, I ran additional sets of simulations with different parameter values (Chapter~\ref{chap:param}).
I showed that these results were robust with regard to the size of the topological domains (the distance at which supercoiling propagates), to the size of the intergenic regions within the range of observed bacterial values, and to the intensity of the environmental perturbations.
Overall, I was able to conclude that the transcription-supercoiling coupling not only provides a material basis for gene regulation in the absence of transcription factors and a mechanism that could explain the activation of certain genes by DNA relaxation, but also plausibly plays a role in the shaping of the organization of bacterial genomes over evolutionary time, through the gene regulatory networks that evolve from genomic rearrangements.

These results, even though they provide fascinating insight into the evolution of bacterial genomes, do however not answer the original question of the putative positive epistatic interactions of supercoiling mutations that was left inconclusively answered in Chapter~\ref{chap:aevol}.
I therefore re-attacked this question using the \emph{EvoTSC} model, by introducing supercoiling mutations along the original genomic inversions (Chapter~\ref{chap:epistasis}).
This time, I investigated whether allowing supercoiling to mutate along genomic inversions would provide a boost in the fitness recovery speed of a population after an environmental shock, but did similarly not detect a meaningful impact.
These negative results ultimately lead to questioning the viability of supercoiling as a fruitful example for the study of the evolutionary role of epistatic interactions.

\section{Limits and Perspectives}

The work presented in this manuscript could be furthered along at least two main directions.
The first one would be to continue investigating supercoiling as a representative example of the role of epistatic interactions in directing evolution.
To that end, several undertakings are actionable.
A first possibility would be to strengthen the analysis of evolutionary trajectories in the \emph{EvoTSC} model, by recording individual lineages during evolution and explicitly reconstructing mutational histories.
This would allow the same experiment as presented in \emph{Aevol} in chapter~\ref{chap:aevol}, to be performed in a model in which supercoiling mutations may have a less determined effect on the fitness landscape, and hence on the reachable evolutionary trajectories.
Studying the fixation times of mutations in \emph{EvoTSC} could provide easier data to interpret than in \emph{Aevol}, as there are only two possible kinds of mutations (genomic inversions and supercoiling mutations) in the model.
Another natural, but slightly more difficult to carry out, direction would be to instead implement the more precise model of supercoiling used in \emph{EvoTSC} in \emph{Aevol}.
Replicating the results already obtained in \emph{EvoTSC} in \emph{Aevol}, a model that is not purpose-built for the study of supercoiling, would first provide an additional degree of confidence in these results.
Reimplementing a model of the transcription-supercoiling coupling in \emph{Aevol} would furthermore allow us to provide a more detailed model of the transcriptional response to supercoiling than in \emph{EvoTSC}.
Incorporating the effect of promoter discriminator sequence and spacer length on transcription initiation~\citep{forquet2021,pineau2022a} is possible in \emph{Aevol} thank to its nucleotide-level description of the genome, and thereby opens additional ways for genomes to evolve in response to supercoiling mutations.
The common thread at the root of both initiatives is to build models in which supercoiling mutations could sufficiently alter the fitness landscape to guide evolution on different paths conditionally to their occurrence in a lineage, and then determine whether they observably do so.
Whether in \emph{EvoTSC} or in \emph{Aevol}, it would therefore seem promising to explore this direction more quantitatively, by computing the distribution of fitness effects of supercoiling mutations in either model, and the fitness of double mutants, instead of relying on the random occurrence of mutations in evolving populations.
This more local but more detailed analysis of the fitness landscape could shed further light on the alleged slant of supercoiling mutations towards beneficial epistasis.

The second main direction in which to pursue the work presented in this manuscript is towards a quantitative description of the regulatory role of the transcription-supercoiling coupling at the mesoscale of bacterial topological domains.
One of the main drawbacks of the model of the interaction between transcription and supercoiling used in \emph{EvoTSC} is its lack of an explicit time scale.
As such, it therefore models the expected average behavior of genes subjected to this interaction, but cannot adequately picture the stochastic nature of gene transcription by RNA polymerases.
Yet, this stochasticity plays an important role in the accurate modeling of gene transcription dynamics, as exemplified in~\cite{sevier2021}.
Incorporating an explicit description of the movement of polymerases along DNA, of supercoil resolution by topoisomerases, and of barrier formation by nucleoid-associated proteins into \emph{EvoTSC} would provide us with a model with which to disentangle the contribution of these processes to gene transcription levels, expanding on initial work such as~\cite{elhoudaigui2019}.
In order to be able to provide quantitative predictions of gene expression, a more precise model of promoter sensitivity to supercoiling would also be needed.
Recent work indeed suggests that the sequence of promoters~\citep{forquet2021}, as well as the length of their spacer~\citep{forquet2022}, are important determinants of the modulation of gene expression by supercoiling.
Coupled with the evolutionary model of \emph{EvoTSC}, such a quantitative model would allow the study of the transcriptional response of genetic systems to perturbations caused by mutations in their components, be it gene order, promoter sequence, or topoisomerase activity, and hence the prevision of possible future evolutionary trajectories.
