\chapter{Introduction}
\label{chap:intro}

The evolution of living things by natural selection is often presented as an unpredictable process, because it originates in the random mutations that affect the core of living things: the DNA molecule, whose sequence is the main carrier of biological information.
However, while it is not possible to identify with certainty which specific mutations will occur in response to a given selection pressure, many experiments suggest that the path that evolution follows is not entirely random and may even be reproducible.
This may seem obvious at the scale of organisms, thinking of the many plants and animals selected and domesticated by the human species over tens of thousands of years, or the repeated appearance of resistance to treatments for bacterial or viral infections.
However, it is only since the 20th century that we have been able to try to understand the basis of this repeatability at the molecular level.
We then observed that it is sometimes the same gene, or even the same nucleotide within a gene, that is affected by mutations when a selection experiment is repeated for a given characteristic.
In this case, evolution no longer seems to be able to follow a multitude of different paths to reach the same visible result, but is forced to stick to a well-defined route.

One mechanism that may explain this repeatability of evolution is epistasis, or the role that the genetic environment plays in the effect of a given mutation.
Indeed, it is possible for a mutation to have a favorable effect in the presence of another mutation, but an unfavorable effect in the absence of it.
These epistatic relationships can thus constrain the options available to evolution, by requiring that a mutation of a given gene occur before that of a second gene, in order for the latter to be favorable.
In a context of competition between different strains within the same population (for example, of pathogenic bacteria), a better understanding of these epistatic relationships would then allow, for example, a more accurate prediction of the fixation or not of future mutations, and thus of the victorious strain, offering the possibility of more accurately orienting a treatment.
The type of epistatic relationships most often studied is that of interactions between point mutations, i.e. between mutations affecting only a small number of contiguous nucleotides within the same gene, as these are the easiest mutations to detect.
However, there are many other types of epistatic interactions that are more complex and less well studied.
For example, the duplication of a gene followed by a divergence and specialization of its two copies in two different functions, a process that plays a major role in innovation at the molecular level, can be interpreted in this context.
There is then an epistasis between a chromosomal rearrangement, the duplication of a gene and the subsequent point mutations; in the absence of this duplication, the mutations which make possible the specialization of the copies of the gene would indeed be deleterious. \textbf{Is the example really necessary?}

The starting point of my thesis was therefore the study of a particular case of epistatic interactions, those generated by mutations in genes regulating DNA supercoiling.
DNA supercoiling, or the degree to which DNA is wrapped around itself, plays an important role in the regulation of gene activity in bacteria, because the level of gene transcription depends directly on the supercoiling at the promoter.
As the level of supercoiling is finely regulated by the activity of several enzymes, called topoisomerases, a mutation in a gene coding for one of these can lead to a change in transcriptional activity at the scale of the entire genome,
This opens the door to the emergence of numerous compensatory mutations.
The evolutionary role of supercoiling mutations, and their repeatability, has been highlighted in particular in the \emph{Long Term Evolution Experiment} carried out in the laboratory of Richard Lenski, which has been evolving 12 strains of \emph{Escherichia coli} in a laboratory environment since 1988.
Indeed, 11 of the 12 strains have seen their supercoiling increase very early in the course of the experiment.

To address the study of the evolutionary role of these mutations during adaptation to a new environment, I opted for an evolutionary systems biology approach.
I started by integrating a gene expression model taking into account the level of supercoiling at the chromosome level in an evolutionary simulation software existing within my thesis team, the software \emph{Aevol}.
This model, and the first results obtained with it, are presented in chapter~\ref{chap:aevol} of the thesis.
However, the experiments carried out in this framework did not lead to a convincing result, the supercoiling converging very quickly during the evolution to a constant level, while the rest of the genome of the individuals continues to evolve.

The role in the regulation of gene expression that the level of supercoiling in bacterial DNA holds actually comes from the highly dynamic nature of supercoiling, which is largely related to gene transcription. \textbf{??}
When a gene is transcribed by an RNA polymerase, the resulting bulky enzyme complex is unable to turn around the DNA as fast as the DNA turns around itself, resulting in an accumulation of supercoiling downstream of the gene, and a deficit of supercoiling upstream of it.
The transcription of a given gene can thus influence, through the changes in supercoiling in its vicinity that it generates, the transcription of genes in its vicinity, thus creating a network of interactions and couplings between the expression levels of nearby genes on the genome.
As such a detailed modeling of supercoiling was difficult to implement in the model \emph{Aevol}, I opted to study, in a first step, its evolutionary role in a model integrating a simplified genome, but describing more accurately supercoiling.
I implemented this model, called \emph{EvoTSC}, from scratch in Python, and it is available at \url{https://gitlab.inria.fr/tgrohens/evotsc}.

Using \emph{EvoTSC}, I first showed that, in a model where the only mechanism for regulating gene activity is supercoiling-mediated coupling between the transcript levels of nearby genes, and where the only possible mutations are chromosomal inversions (which rearrange the relative positions of the genes), it is possible to obtain by natural selection individuals whose genes accurately follow environment-dependent targets.
In particular, it is possible to obtain genes that are activated by global DNA relaxation, while their promoters are inhibited by local relaxation.
These first results are presented in the chapter~\ref{chap:alife}, and demonstrate that supercoiling can indeed play a major role in the regulation of bacterial gene activity, as a support of a genetic regulation network.
They were published, first as a paper in the conference \emph{ALIFE 2021}~\citep{grohens2021}, and then in an extended version in the associated journal, \emph{Artificial Life}~\citep{grohens2022a}.

In a second step, I then sought to characterize in more detail the evolutionary impact of supercoiling in the structure of bacterial genomes.
Still using the \emph{EvoTSC} model, I showed that at the most local level, convergent or divergent pairs of neighboring genes are formed, in accordance with theoretical predictions of the coupling between supercoiling and transcription.
I have shown that this organization at the local genome scale is, however, not entirely sufficient to explain the gene expression levels observed in the whole genome, but that subnetworks involving up to several tens of genes may instead be required.
Finally, using a gene knockout approach, I have shown that in the genomes of evolved individuals, it is in the form of a single, genome-wide network that the regulation of gene expression is organized in the EvoTSC model.
This second set of results is presented in chapter~\ref{chap:ploscb}, and has been formatted in a pre-publication~\citep{grohens2022b}, soon to be submitted for peer review.

In chapter~\ref{chap:param}, I then present a set of complementary experiments that show the robustness of the model results \emph{EvoTSC} presented so far, when varying the main parameters of the model.
Finally, I have incorporated in EvoTSC a model of the evolution of the global supercoiling level, in order to characterize, as in the experiments conducted with Aevol, the possible epistatic relationships between supercoiling mutations and chromosomal rearrangements.
These results are presented in the chapter~\ref{chap:epistasis}.

The appendix~\ref{chap:software} presents the software contributions I made throughout my thesis.
I first participated in the development of \emph{Aevol} and associated tools to manage simulations, then developed the \emph{EvoTSC} model as well as a set of tools to visualize and analyze the resulting data.

Finally, the course of my thesis having been disrupted by the outbreak of the Covid-19 pandemic in France in the spring of 2020, I volunteered to collaborate, within a team of Inria researchers and engineers, with the Assistance Publique-HÃ´pitaux de Paris (AP-HP).
Together we built a model of the epidemic in the Parisian agglomeration, which aimed to help the AP-HP teams to follow in real time and try to predict the evolution of the epidemic using medical regulation data.
This work subsequently led to a publication~\citep{gaubert2020}, presented in Appendix~\ref{chap:covid}.

\selectlanguage{english}
